src/reference-manual/user-functions.Rmd

```diff
@@ -223,24 +227,52 @@ real foo_lpdf(real y, vector theta) { ... }
 ```
 
 Note that for function definitions, the comma is used rather than the
-vertical bar.  Then the shorthand
+vertical bar.
+
+For every custom `_lpdf` and `_lpmf` defined there is a corresponding
+`_lupdf` and `_lupmf` defined automatically. The `_lupdf` and `_lupmf`
+versions of the functions cannot be defined directly (to do so will
+produce an error). The difference in the
+`_lpdf` and `_lpmf` and the corresponding `_lupdf` and `_lupmf` functions
+is that if any other unnormalized density functions are used
+inside the user-defined function, the `_lpdf` and `_lpmf` forms of the
+user-defined function will change these densities to be normalized.
+The `_lupdf` and `_lupmf` forms of the user-defined functions will
+instead allow other unnormalized density functions to drop additive
+constants.
+
+The sampling shorthand
 
 ```stan
 z ~ foo(phi);
 ```
 
-will have exactly the same effect
+will have the same effect as incrementing the target with the log
+of the unnormalized density:
 
 ```stan
+target += foo_lupdf(z | phi);
+```
+
+Other `_lupdf` and `_lupmf` functions used in the definition of
+`foo_lpdf` will drop additive constants. If there are no `_lupdf`
```

Comment:
I might just be confused (as is often the case!) but it says just above that 

> if any other unnormalized density functions are used inside the user-defined function, the `_lpdf` and `_lpmf` forms of the user-defined function will change these densities to be normalized

Does that contradict this?

Reply:
I think this is right but I rewrote it to make it clearer.